### PR TITLE
Cleanup whether to enter user namespace

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes problems found in https://github.com/containers/libpod/issues/5306

File: cmd/podman/main_local.go [hunk 1]
```diff
@@ -243,6 +243,27 @@ func setupRootless(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func useUserNS(cmd *cobra.Command) bool {
+	if os.Geteuid() == 0 {
+		return false
+	}
+	switch cmd {
+	case _migrateCommand,
+		_migrateCommand,
```
Comment (vrothberg) [inline cmd/podman/main_local.go:252]:
```suggestion
```

File: cmd/podman/main_local.go [hunk 2]
```diff
@@ -243,6 +243,27 @@ func setupRootless(cmd *cobra.Command, args []string) error {
...
+		_infoCommand,
+		_searchCommand,
+		_systemResetCommand,
+		_versionCommand:
+		return false
+	}
+	if strings.HasPrefix(cmd.Use, "help") {
+		return false
+	}
+	return true
```
Comment (vrothberg) [inline cmd/podman/main_local.go:264]:
```suggestion
	return !strings.HasPrefix(cmd.Use, "help")
```

Comment (vrothberg) [reply cmd/podman/main_local.go:264]: Can you add a comment to this check? As far as I understand it, we are checking if the "Use" string of the command starts with "help" but there are no such occurrences in the code:
```bash
$ grep "Use:" -r cmd/ | grep help | wc -l
0
```

**Comment by giuseppe** [comment]
I am afraid we won't be able to rm files not owned by the rootless user without entering the user namespace.

**Comment by rhatdan** [comment]
Yes I just thought of the same thing.

File: cmd/podman/main_local.go (hunk 2 above)
Comment (rhatdan) [reply cmd/podman/main_local.go:264]: I think this was added for commands like

podman run --help

**Comment by rhatdan** [comment]
This PR no longer fixes the issue, but I think it cleans up the code and adds

File: cmd/podman/main_local.go (hunk 1 above)
Comment (rhatdan) [reply cmd/podman/main_local.go]: Fixed

File: cmd/podman/main_local.go (hunk 2 above)
Comment (rhatdan) [reply cmd/podman/main_local.go]: Done.

Comment (vrothberg) [reply cmd/podman/main_local.go]: `Cmd.Use` is the wrong field for that:
```golang
// Command is just that, a command for your application.              
// E.g.  'go run ...' - 'run' is the command. Cobra requires          
// you to define the usage and description as part of your command    
// definition to ensure usability.                                    
type Command struct {                                                 
        // Use is the one-line usage message.                         
        Use string                                                    
```

Comment (vrothberg) [reply cmd/podman/main_local.go]: I printf-debuged this code path. If we run `--help` we don't execute the code path. I guess we can just `return true`.

Comment (rhatdan) [reply cmd/podman/main_local.go]: Nice, so I guess this code is not needed, removed.

File: cmd/podman/main_local.go [hunk 3]
```diff
@@ -243,6 +243,22 @@ func setupRootless(cmd *cobra.Command, args []string) error {
...
+	case _migrateCommand,
+		_mountCommand,
+		_renumberCommand,
+		_infoCommand,
+		_searchCommand,
+		_versionCommand:
+		return false
+	}
+	return true
+}
```
Comment (TomSweeneyRedHat) [inline cmd/podman/main_local.go:260]: I'd like a better function name and/or a function comment over what this function is supposed to do.

**Comment by mheon** [comment]
One question: can we still remove containers without entering the user namespace? We may need to delete files not owned by the current user.

I was thinking we needed a way to enter the userns without joining the pause process.

File: cmd/podman/main_local.go (hunk 3 above)
Comment (rhatdan) [reply cmd/podman/main_local.go:260]: Suggestions?

Comment (rhatdan) [reply cmd/podman/main_local.go:263]: Changed name to executeCommandInUserNS
and added a comment.

**Comment by giuseppe** [comment]
> I was thinking we needed a way to enter the userns without joining the pause process.

We have a fallback in place for that.  When the pause process cannot be joined in some cases it is enough to just create a new namespace.  Looks like something is going wrong there when not using systemd

**Pull request merged**